test-runner-poc/lib/runner.js

```diff
@@ -2,72 +2,122 @@
 
 const path = require("path");
 const fs = require("fs");
-const utils = require("mocha").utils;
+const MongoClient = require('mongodb').MongoClient;
+const environments = require('../environments');
+const TestConfiguration = require('../config');
+const parseConnectionString = require('../../lib/core/uri_parser');
+const mock = require('mongodb-mock-server');
 
+let mongoClient;
 let filters = [];
 let files = [];
 
 function addFilter(filter) {
-	if (typeof filter !== "function" && typeof filter !== "object") {
-		throw new Error(
-			"Type of filter must either be a function or an object"
-		);
-	}
-	if (
-		typeof filter === "object" &&
-		(!filter.filter || typeof filter.filter !== "function")
-	) {
-		throw new Error("Object filters must have a function named filter");
-	}
-
-	if (typeof filter === "function") {
-		filters.push({ filter: filter });
-	} else {
-		filters.push(filter);
-	}
+  if (typeof filter !== 'object') {
+    throw new Error('Type of filter must be an object');
+  }
+  if (!filter.filter || typeof filter.filter !== 'function') {
+    throw new Error('Object filters must have a function named filter');
+  }
+  filters.push(filter);
+
+}
+function environmentSetup(environmentCallback) {
+  const mongodb_uri = process.env.MONGODB_URI || 'mongodb://localhost:27017';
```

Comment:
you are creating a `MongoClient` and connecting it for envrionment setup, as well as the mongodb version filter. Should we instead create a single client, and pass it into these methods?

Reply:
So create a `MongoClient` and pass it into those methods without connecting it?

Reply:
I was thinking make a common `MongoClient`, connect it, and then pass it into various things that need to work with it